framework/TestFiles.py

```diff
@@ -0,0 +1,244 @@
+  if update:
+    if res:
+      results["pass"] += 1
+    else:
+      print("checking error",comment,'|',msg)
+      results["fail"] += 1
+  print(' ... end Error testing')
+  print('')
+  return res
+
+# TODO test RAVENGenerated, File (base type), CSV, DynamicXMLOutput, and UserGenerated types
+# These are covered in regression tests, but not yet in unit tests.
+
+######################################
+#          Static XML                #
+######################################
+# create it
+static = Files.StaticXMLOutput()
+static.messageHandler = mh
+# start a new tree
+static.newTree('TestRoot')
+# test instance
+checkTrue('Static root node type',isinstance(static.tree,ET.ElementTree))
+# test tag
+checkSame('Static root node name',static.tree.getroot().tag,'TestRoot')
+# add some scalars
+static.addScalar('FirstTarget','pi',3.14159)
+static.addScalar('FirstTarget','e',6.18)
+# check values
+pi = xmlUtils.findPath(static.tree.getroot(),'FirstTarget/pi')
+e = xmlUtils.findPath(static.tree.getroot(),'FirstTarget/e')
+checkFloat('Static pi',float(pi.text),3.14159)
+checkFloat('Static e',float(e.text),6.18)
+# add a vector
+static.addVector('SecondTarget','pi',{'e':0,'pi':1})
+static.addVector('SecondTarget','e',{'e':1,'pi':0})
+# check values
+checkFloat('Static pi wrt pi',float(xmlUtils.findPath(static.tree.getroot(),'SecondTarget/pi/pi').text),1.0)
+checkFloat('Static pi wrt e',float(xmlUtils.findPath(static.tree.getroot(),'SecondTarget/pi/e').text),0.0)
+checkFloat('Static e wrt pi',float(xmlUtils.findPath(static.tree.getroot(),'SecondTarget/e/pi').text),0.0)
+checkFloat('Static e wrt e',float(xmlUtils.findPath(static.tree.getroot(),'SecondTarget/e/e').text),1.0)
+# test pickling
+p = pk.dumps(static)
+new = pk.loads(p)
+# check all values on new
+checkTrue('Static pk root node type',isinstance(static.tree,ET.ElementTree))
+checkSame('Static pk root node name',static.tree.getroot().tag,'TestRoot')
+pi = xmlUtils.findPath(new.tree.getroot(),'FirstTarget/pi')
+e = xmlUtils.findPath(new.tree.getroot(),'FirstTarget/e')
+checkFloat('Static pk pi',float(pi.text),3.14159)
+checkFloat('Static pk e',float(e.text),6.18)
+checkFloat('Static pk pi wrt pi',float(xmlUtils.findPath(new.tree.getroot(),'SecondTarget/pi/pi').text),1.0)
+checkFloat('Static pk pi wrt e',float(xmlUtils.findPath(new.tree.getroot(),'SecondTarget/pi/e').text),0.0)
+checkFloat('Static pk e wrt pi',float(xmlUtils.findPath(new.tree.getroot(),'SecondTarget/e/pi').text),0.0)
+checkFloat('Static pk e wrt e',float(xmlUtils.findPath(new.tree.getroot(),'SecondTarget/e/e').text),1.0)
+
+# TODO test writing to file
+
+
+print(results)
+
+sys.exit(results["fail"])
+"""
+  <TestInfo>
+    <name>framework.unit_test_Files</name>
+    <author>talbpaul</author>
+    <created>2017-11-01</created>
+    <classesTested>Files</classesTested>
+    <description>
+       This test is a Unit Test for the Files classes.
+    </description>
+  </TestInfo>
+"""
```

Comment:
move this to the top?

Reply:
This precedent has been established and used in all of our RavenPython tests thus far; this is consistent.

Reply:
OK, fine with me.